File: ui/console_interface.py
from services.car_rental_service import CarRentalService

class ConsoleInterface:
    def __init__(self):
        self.service = CarRentalService()

    def start(self):
        print("Bienvenue dans l'application de location de voitures")
        while True:
            self.show_menu()

    def show_menu(self):
        print("\n--- MENU LOCATION VOITURE ---")
        print("1. Voir les voitures disponibles")
        print("2. Louer une voiture")
        print("3. Retourner une voiture")
        print("0. Quitter")

        choice = input("Votre choix : ").strip()

        if choice == "1":
            self.list_cars()
        elif choice == "2":
            self.rent_car()
        elif choice == "3":
            self.return_car()
        elif choice == "0":
            print("Merci pour votre visite !")
            exit()
        else:
            print("Choix invalide.")

    def list_cars(self):
        print("\nListe des voitures :")
        for car in self.service.list_cars():
            print(car)

    def rent_car(self):
        car_id = input("Entrez l'ID de la voiture à louer : ").strip()
        car = self.service.rent_car(car_id)
        if car:
            print("Vous avez loué :", car)
        else:
            print("Voiture non disponible ou ID incorrect.")

    def return_car(self):
        car_id = input("Entrez l'ID de la voiture à retourner : ").strip()
        car = self.service.return_car(car_id)
        if car:
            print("Merci pour le retour :", car)
        else:
            print("ID incorrect ou voiture déjà disponible.")

File: ui/ConsoleInterface.js
import promptSync from 'prompt-sync';
import CarRentalService from '../services/CarRentalService.js';

export default class ConsoleInterface {
    constructor() {
        this.service = new CarRentalService();
        this.prompt = promptSync();
    }

    start() {
        console.log('Bienvenue dans l\'application de location de voiture');
        this.showMenu();
    }

    showMenu() {
        console.log('\n--- MENU LOCATION VOITURE ---');
        console.log('1. Voir les voitures disponibles');
        console.log('2. Louer une voiture');
        console.log('3. Retourner une voiture');
        console.log('0. Quitter');

        const choice = this.prompt('Votre choix : ').trim();
        switch (choice) {
            case '1':
                this.listCars();
                break;
            case '2':
                this.rentCar();
                break;
            case '3':
                this.returnCar();
                break;
            case '0':
                console.log('Merci pour votre visite !');
                process.exit(0);
            default:
                console.log('Choix invalide.');
                this.showMenu();
        }
    }

    listCars() {
        const cars = this.service.listCars();
        console.log('\nListe des voitures :');
        cars.forEach(car => console.log(car.toString()));
        this.showMenu();
    }

    rentCar() {
        const id = this.prompt("Entrez l'ID de la voiture à louer : ").trim();
        const car = this.service.rentCar(id);
        if (car) {
            console.log('Vous avez loué : ' + car.toString());
        } else {
            console.log('Voiture non disponible ou ID incorrect.');
        }
        this.showMenu();
    }

    returnCar() {
        const id = this.prompt("Entrez l'ID de la voiture à retourner : ").trim();
        const car = this.service.returnCar(id);
        if (car) {
            console.log('Merci pour le retour : ' + car.toString());
        } else {
            console.log('ID incorrect ou voiture déjà disponible.');
        }
        this.showMenu();
    }
}
